src/librustc_trait_selection/traits/coherence.rs

```diff
@@ -389,7 +389,7 @@ fn orphan_check_trait_ref<'tcx>(
     ) -> Vec<Ty<'tcx>> {
         // FIXME(eddyb) figure out if this is redundant with `ty_is_non_local`,
         // or maybe if this should be calling `ty_is_non_local_constructor`.
```

Comment:
This `FIXME` is interesting.

Will have to do something about this before we merge this PR

Reply:
Looking into the current impl it seems somewhat overly complicated but can't quite think of a clear solution here.

I do feel like using a `TypeWalker` in `fn orphan_check_trait_ref` might actually be the cleanest approach here, but iirc we wanted to slowly phase out uses of this struct.

What's the current status here?

Reply:
I really would like to clean this up before merging this.

If we merge it without fixing this, we should at least update the method names :sweat_smile: